src/node_api.cc

```diff
@@ -105,23 +114,40 @@ namespace {
 
 class BufferFinalizer : private Finalizer {
  public:
+  static BufferFinalizer* New(napi_env env,
+                              napi_finalize finalize_callback = nullptr,
+                              void* finalize_data = nullptr,
+                              void* finalize_hint = nullptr) {
+    return new BufferFinalizer(
+        env, finalize_callback, finalize_data, finalize_hint);
+  }
   // node::Buffer::FreeCallback
   static void FinalizeBufferCallback(char* data, void* hint) {
     std::unique_ptr<BufferFinalizer, Deleter> finalizer{
         static_cast<BufferFinalizer*>(hint)};
-    finalizer->_finalize_data = data;
+    finalizer->finalize_data_ = data;
 
-    if (finalizer->_finalize_callback == nullptr) return;
-    finalizer->_env->CallFinalizer(finalizer->_finalize_callback,
-                                   finalizer->_finalize_data,
-                                   finalizer->_finalize_hint);
+    // It is same to call into JavaScript at this point.
```

Comment:
typo? "safe"

Reply:
@legendecas this looks like something that still needs to be fixed.